File: experiment/java-python/test/j2pyrawpy/for__0__COUNT_DISTINCT_SUBSEQUENCES.py
#!/usr/bin/env python
""" generated source for module for__0__COUNT_DISTINCT_SUBSEQUENCES """
class X(object):
    """ generated source for class X """
    @classmethod
    def countSub(cls, str_):
        """ generated source for method countSub """
        last = [None]*MAX_CHAR
        Arrays.fill(last, -1)
        n = len(str_)
        dp = [None]*n + 1
        dp[0] = 1
        return dp[n]

File: experiment_jp_test/j2pyraw/for__0__COUNT_DISTINCT_SUBSEQUENCES.java
class X { static int countSub ( String str ) { int [ ] last = new int [ MAX_CHAR ] ; Arrays . fill ( last , - 1 ) ; int n = str . length ( ) ; int [ ] dp = new int [ n + 1 ] ; dp [ 0 ] = 1 ; return dp [ n ] ; }
 }